ai/battery.py

```diff
@@ -0,0 +1,170 @@
+from bot_utils import get_id
+from discord.ext import tasks, commands
+from db.drone_dao import is_drone, is_battery_powered, deincrement_battery_minutes_remaining, get_battery_percent_remaining, get_all_drone_batteries, get_battery_minutes_remaining, set_battery_minutes_remaining
+from id_converter import convert_ids_to_members
+import logging
+from resources import MAX_BATTERY_CAPACITY_MINS, DRONE_AVATAR, HOURS_OF_RECHARGE_PER_HOUR
+from roles import has_role, BATTERY_POWERED, BATTERY_DRAINED, HIVE_MXTRESS
+from discord.utils import get
+import webhook
+from ai.identity_enforcement import identity_enforcable
+import emoji
+import re
+
+LOGGER = logging.getLogger('ai')
+
+
+class BatteryCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.draining_batteries = {}  # {drone_id: minutes of drain left}
```

Comment:
```suggestion
        self.draining_batteries: Dict[str, int] = {}  # {drone_id: minutes of drain left}
```
Could be an explicit type hint to provide IDE support
Requires additional import `from typing import Dict`